lib/services/runtime/handlers/message/lib/conditions/conditionIsolate.ts

```diff
@@ -0,0 +1,44 @@
+import ivm from 'isolated-vm';
+
+export class ConditionIsolate {
+  private isolate?: ivm.Isolate;
+
+  private context?: ivm.Context;
+
+  constructor(private readonly variables: Record<string, unknown>) {}
+
+  private readonly ISOLATED_VM_LIMITS = {
+    maxMemoryMB: 8,
+    maxExecutionTimeMs: 1 * 1000,
+  };
+
+  public async initialize(): Promise<void> {
+    this.isolate = new ivm.Isolate({
+      memoryLimit: this.ISOLATED_VM_LIMITS.maxMemoryMB,
+    });
+
+    this.context = await this.isolate.createContext();
+
+    await Promise.all(
+      Object.keys(this.variables).map(async (key) => {
+        await this.context!.global.set(key, this.variables[key], {
+          copy: true,
+        });
+      })
+    );
```

Comment:
```suggestion
    await Promise.all(
      Object.entries(this.variables).map(([key, value]) =>
        this.context?.global.set(key, value, {
          copy: true,
        })
      )
    );
```

1. you're already returning a promise, no need to `async/await`
2. try and avoid using `!` for typings. this may be from typescript not inferring that it is definitely not undefined after the `await` call, but either casting above, or adding a check, or etc, etc, should be done before the use of `!`
3. use `Object.entries` to get both key and value to save a second lookup of the value within the loop